client/app/components/BaseContainer/index.jsx

```diff
@@ -0,0 +1,109 @@
+// @flow
+import React from 'react';
+import { StoryContainer } from './StoryContainer';
+
+type A = { data: any, lastPage: any };
+
+export type Props = {
+  container: string,
+  data: any,
+  fetchUrl: string,
+  lastPage?: boolean,
+};
+
+export type State = {
+  loading: boolean,
+  lastPage: boolean,
+  page: number,
+  data: any,
+};
+
+export class BaseContainer extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      page: 1,
+      lastPage: props.lastPage || false,
+      loading: false,
+      data: props.data,
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('scroll', this.handleScroll);
+  }
+
+  componentDidUpdate() {
+    const { lastPage } = this.state;
+    if (lastPage) {
+      window.removeEventListener('scroll', this.handleScroll);
+    }
+  }
+
+  makeCall = (url: string, page: number): Promise<A> => new Promise((resolve: any, reject) => {
+    const xhr = new XMLHttpRequest();
```

Comment:
Let's actually use Axios for HTTP requests. [Here's](https://github.com/ifmeorg/ifme/blob/master/client/app/widgets/Comments/index.jsx#L43) an example file we use.